## Online Stereo Camera VTR3

session_name: online_vtr_camera

# environment:
#   ROS_DOMAIN_ID: "1"  # set this to a unique number when multiple ROS2 processes running on the same network

start_directory: ${VTRDATA}

suppress_history: false

windows:
  - window_name: vtr_main
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/main/install/setup.bash
    panes:
      - >
        sleep 3 &&
        ros2 launch vtr_navigation vtr.launch.py
        base_params:=bumblebee_zeus_default.yaml
        data_dir:=${VTRDATA}/online-test-camera/zeus/$(date '+%F')/$(date '+%F')_main
        clear_data_dir:=false

      - sleep 5 && ros2 run rviz2 rviz2 -d ${VTRSRC}/rviz/camera.rviz

  - window_name: vtr_ui
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/main/install/setup.bash
    panes:
      - sleep 3 && ros2 run vtr_ui socket_client  --ros-args -r __ns:=/vtr # wait for the socket_server to be created
      - ros2 run vtr_ui web_server  --ros-args -r __ns:=/vtr
      - ros2 run vtr_ui socket_server  --ros-args -r __ns:=/vtr
      - sleep 5 && firefox --new-window "localhost:5200" # the webpage has to wait for everything above

  - window_name: zeus
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/robots/ros2/install/setup.bash
    panes:
      - ros2 launch zeus_description zeus_description.launch.py

  - window_name: bumblebee
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/extra/install/setup.bash
    panes:
      - ros2 run vtr_bumblebee_xb3 BumblebeeDriver --ros-args --params-file ${VTRSRC}/extra/src/vtr_bumblebee_xb3/param/XB3.yaml
      - stress -c 1 # workaround for super weird XB3 Driver bug
